### PR TITLE
Add support for Redis Cache VNET integration

### PR DESCRIPTION
**Pull request by v4ld3r5**
This PR solves #402, adds support to create a fully private Redis Cache:
* Adds `public_network_access_enabled` parameter to module's interface
* Sets new module variable to pass the `subnet.id` directly, instead of part of `redis` map, to be able to obtain it dynamically from a landing zone
* Add `subnet_id` parameter from top module and construct the lookup from `combined_objects`
* Depend on networking module for ordering purposes

This code has been tested and I'm able to successfully deploy a Premium instance within vnet.

### CODE REVIEW

File: redis_caches.tf [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
+  depends_on = [module.networking]
+  for_each   = local.database.azurerm_redis_caches
 
   tags                = try(each.value.tags, null)
   location            = lookup(each.value, "region", null) == null ? module.resource_groups[each.value.resource_group_key].location : local.global_settings.regions[each.value.region]
   resource_group_name = module.resource_groups[each.value.resource_group_key].name
   redis               = each.value.redis
   global_settings     = local.global_settings
   base_tags           = try(local.global_settings.inherit_tags, false) ? module.resource_groups[each.value.resource_group_key].tags : {}
+  subnet_id           = try(each.value.subnet_key, null) == null ? null : try(local.combined_objects_networking[local.client_config.landingzone_key][each.value.vnet_key].subnets[each.value.subnet_key].id, local.combined_objects_networking[each.value.lz_key][each.value.vnet_key].subnets[each.value.subnet_key].id)
```
Comment (LaurentLesle) [inline redis_caches.tf:13]: USe the simplified structure

**Review by LaurentLesle — CHANGES_REQUESTED** [review]
Thanks for the PR. I put some comments that are consistent across the 3 PR you have sumbitted.

Add an example here - https://github.com/aztfmod/terraform-azurerm-caf/tree/master/examples/redis_cache

Reference the example in the pipelines https://github.com/aztfmod/terraform-azurerm-caf/tree/master/.github/workflows

File: redis_caches.tf (hunk 1 above)
Comment (v4ld3r5) [reply redis_caches.tf:13]: Hi @LaurentLesle , please find the additional commit with the example and reference

Br

Comment (v4ld3r5) [reply redis_caches.tf:13]: > USe the simplified structure

Is there a more simpler way? I couldn't find it in the codebase. Do you mind giving an example?

Comment (LaurentLesle) [reply redis_caches.tf:13]: I was referring to the following:

`subnet_id = try(local.combined_objects_networking[try(each.value.lz_key, local.client_config.landingzone_key)][each.value.vnet_key].subnets[each.value.subnet_key].id, null)` 

Try and if it works you that in the other PR you have submitted

Comment (v4ld3r5) [reply redis_caches.tf]: Thx got it, updated

**Review by v4ld3r5 — COMMENTED** [review]
Corrected example folder name, added pipeline links

**Pull request merged**